### ISSUE TITLE
TcpReceiveSendGetsCanceledByDispose test fails on rhel7

### ISSUE THREAD

**Issue by tmds** [dotnet/runtime#52597]
These `TcpReceiveSendGetsCanceledByDispose` tests fail on our internal CI server on rhel7:
```
System.Net.Sockets.Tests.SendReceive_SpanSyncForceNonBlocking.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: False, dualModeClient: True)
System.Net.Sockets.Tests.SendReceive_SpanSyncForceNonBlocking.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: True, dualModeClient: False)
System.Net.Sockets.Tests.SendReceive_Sync.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: False, dualModeClient: True)
System.Net.Sockets.Tests.SendReceive_Sync.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: True, dualModeClient: False)
System.Net.Sockets.Tests.SendReceive_SyncForceNonBlocking.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: False, dualModeClient: True)
System.Net.Sockets.Tests.SendReceive_SyncForceNonBlocking.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: True, dualModeClient: False)
System.Net.Sockets.Tests.SendReceive_SpanSync.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: False, dualModeClient: True)
System.Net.Sockets.Tests.SendReceive_SpanSync.TcpReceiveSendGetsCanceledByDispose(receiveOrSend: True, ipv6Server: True, dualModeClient: False)
```
The tests that are failing are matching this condition:

https://github.com/dotnet/runtime/blob/5a47690c4658580379a4a9cf9a375bad6bc27949/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs#L1002-L1005

The failure exception is masked by the Timeout (`[Theory(Timeout = 40000)]`) that is applied to this test:
```xml
        <failure exception-type="Xunit.Sdk.TestTimeoutException">
          <message><![CDATA[Test execution timed out after 40000 milliseconds]]></message>
          <stack-trace><![CDATA[]]></stack-trace>
        </failure>
```

cc @antonfirsov

**Comment by tmds** [comment]
cc @omajid

**Comment by tmds** [comment]
I have debugged this further.

This looks like a regression in the RHEL kernel. I have installed RHEL 7.9 in a VM which came with kernel `3.10.0-1160.el7.x86_64`. Tests passed.
I updated my system and kernel changed to `3.10.0-1160.25.1.el7.x86_64` and now tests fail.

I'll look at writing a reproducer in C and when it fails report it as a kernel bug.

When I got the reproducer, I'd like to make a PR to skip this combination. Otherwise our internal CI will fail until there is a patched kernel. The public CI will also fail when/if it moves to the latest RHEL7 kernel.
